### PR TITLE
Make private jobs to use the oauth volume defined in decoration config

### PR DESCRIPTION
**Pull request by droslean**
/cc @openshift/test-platform 

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

**Comment by droslean** [comment]
follow-up https://github.com/openshift/release/pull/21394

**Comment by droslean** [comment]
/hold

This change needs 
- Bump prow to latest which allows this kind of validation
- Follow-up in o/release to regenerate the private jobs.

**Comment by droslean** [comment]
@stevekuznetsov This PR is ready for review.

File: pkg/prowgen/prowgen_test.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ func TestGeneratePodSpec(t *testing.T) {
 		secrets        []*ciop.Secret
 		targets        []string
 		additionalArgs []string
+		shouldClone    bool
```
Comment (stevekuznetsov) [inline pkg/prowgen/prowgen_test.go:25]: invert this to make it easier to read - we always consider "skipCloning"

File: pkg/prowgen/prowgen_test.go [hunk 2]
```diff
@@ -54,11 +55,20 @@ func TestGeneratePodSpec(t *testing.T) {
 			},
 			targets: []string{"target"},
 		},
+		{
+			description: "private job with skip_cloning false",
```
Comment (stevekuznetsov) [inline pkg/prowgen/prowgen_test.go:59]: test case for the opposite?

File: pkg/prowgen/prowgen_test.go [hunk 3]
```diff
@@ -726,6 +736,17 @@ func TestGenerateJobBase(t *testing.T) {
 			podSpec: &corev1.PodSpec{Containers: []corev1.Container{{Name: "test"}}},
 			clone:   true,
 		},
+		{
+			testName: "private jobs without skipcloning should contain oauth_token_secret config",
```
Comment (stevekuznetsov) [inline pkg/prowgen/prowgen_test.go:740]: don't use double negatives, just harder to read

```
private jobs that clone should ...
```

File: test/integration/ci-operator-prowgen/output/jobs/private-org/super/private-org-super-master-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+presubmits:
```
Comment (stevekuznetsov) [inline test/integration/ci-operator-prowgen/output/jobs/private-org/super/private-org-super-master-presubmits.yaml:1]: test for skip cloning?

Comment (droslean) [reply test/integration/ci-operator-prowgen/output/jobs/private-org/super/private-org-super-master-presubmits.yaml:1]: We should already have those tests. see [here](https://github.com/openshift/ci-tools/blob/master/test/integration/ci-operator-prowgen/output/jobs/private-org/duper/private-org-duper-master-presubmits.yaml#L1-L58)

File: pkg/prowgen/prowgen_test.go (hunk 2 above)
Comment (droslean) [reply pkg/prowgen/prowgen_test.go:59]: The `private job` test case is the opposite.

**Pull request merged**